tests/integration_test.py

```diff
@@ -0,0 +1,360 @@
+        # or 30 if deletions and updates apply to different records.
+        diff_since_last = preview_client.get_records(_since=preview_timestamp)
+        assert (
+            25 <= len(diff_since_last) <= 30
+        ), "Changes since last signature are not consistent"
+
+        metadata = preview_client.get_collection()["data"]
+        assert (
+            preview_signature != metadata["signature"]
+        ), "Preview collection not updated"
+
+    # 2.3 approve the review
+    print("Reviewer approves and triggers signature")
+    data = {"status": "to-sign"}
+    reviewer_client.patch_collection(data=data)
+
+    # 5. wait for the result
+
+    # 6. obtain the destination records and serialize canonically.
+
+    records = list(dest_client.get_records())
+    assert len(records) == expected, f"{len(records)} != {expected} records"
+    timestamp = collection_timestamp(dest_client)
+    serialized = canonical_json(records, timestamp)
+
+    # 7. get back the signed hash
+
+    signature = dest_client.get_collection()["data"]["signature"]
+
+    with open("pub", "w") as f:
+        f.write(signature["public_key"])
+
+    # 8. verify the signature matches the hash
+    signer = ECDSASigner(public_key="pub")
+    try:
+        signer.verify(serialized, signature)
+        print("Signature OK")
+    except Exception:
+        print("Signature KO")
+        raise
+
+
+def test_changes_plugin(get_clients: Tuple[Client, Client, Client]):
+    client, _, _ = get_clients
+    client.create_bucket(id="blog", if_not_exists=True)
+    client.create_collection(id="articles", bucket="blog", if_not_exists=True)
+    records = client.get_records(bucket="monitor", collection="changes")
+
+    assert records
+    assert len(records) == 1
+    assert "bucket" in records[0]
+    assert records[0]["bucket"] == "blog"
+
+    initial_last_modified = records[0]["last_modified"]
+
+    upload_records(client, 10, "blog", "articles")
+    records = client.get_records(bucket="monitor", collection="changes")
+
+    updated_last_modified = records[0]["last_modified"]
+
+    assert updated_last_modified > initial_last_modified
+
+
+def _rand(size: int = 10) -> str:
+    return "".join([random.choice(hexdigits) for _ in range(size)])
+
+
+def collection_timestamp(client: Client) -> str:
+    # XXXX Waiting https://github.com/Kinto/kinto-http.py/issues/77
+    endpoint = client.get_endpoint("records")
+    _, headers = client.session.request("get", endpoint)
+    return headers.get("ETag", "").strip('"')
+
+
+def upload_records(
+    client: Client, num: int, bucket: str = None, collection: str = None
+) -> List[Dict]:
+    records = []
+    for _ in range(num):
+        data = {"one": _rand(1000)}
+        record = client.create_record(data=data, bucket=bucket, collection=collection)
+        records.append(record["data"])
+    return records
+
+
+def test_admin_login(base_url: str, selenium: WebDriver):
```

Comment:
Wouldn't we want to have this in a separate file? (eg. `browser_test.py` ?)

Reply:
https://github.com/mozilla-services/kinto-dist/pull/1953/commits/9644e09acba2074bec8483b5b356b9bbc01da5bd

Will add more tests.